### PR TITLE
update container config.toml to use v2 shim

### PR DESCRIPTION
**Pull request by eriksywu**

### CODE REVIEW

File: parts/linux/cloud-init/nodecustomdata.yml [hunk 1]
```diff
@@ -266,24 +266,24 @@ write_files:
     [plugins."io.containerd.grpc.v1.cri"]
       sandbox_image = "{{GetPodInfraContainerSpec}}"
       [plugins."io.containerd.grpc.v1.cri".containerd]
-        {{ if TeleportEnabled }}
+        {{- if TeleportEnabled }}
```
Comment (paulgmiller) [inline parts/linux/cloud-init/nodecustomdata.yml:269]: why touch teleport here? Must admit I am not familiar with when - is needed or not

File: parts/linux/cloud-init/nodecustomdata.yml [hunk 2]
```diff
@@ -266,24 +266,24 @@ write_files:
...
           runtime_engine = "/usr/bin/nvidia-container-runtime"
-          {{- else}}
-          runtime_engine = "/usr/bin/runc"
-          {{- end}}
+        {{ else}}
+        default_runtime_name = "runc"
+        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc]
+          runtime_type = "io.containerd.runc.v2"
+        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.untrusted]
+          runtime_type = "io.containerd.runc.v2"
```
Comment (paulgmiller) [inline parts/linux/cloud-init/nodecustomdata.yml:285]: no runtime_engine  needed for v2? checkin aks-engine settigns.

Comment (paulgmiller) [reply parts/linux/cloud-init/nodecustomdata.yml:285]: https://github.com/Azure/aks-engine/blob/b3aa75c463764f8fb72b13a7beb4752a6a4e1f37/pkg/api/common/const.go#L283

Comment (paulgmiller) [reply parts/linux/cloud-init/nodecustomdata.yml:285]: looks like this is the same as aks-engine so I think I'm signed off except for the teleport touch.

**Review by paulgmiller — APPROVED** [review]
Seems to bring us inline wiht aks-engien

File: parts/linux/cloud-init/nodecustomdata.yml (hunk 2 above)
Comment (cpuguy83) [reply parts/linux/cloud-init/nodecustomdata.yml:288]: Yeah I think its best to be explicit here.

File: pkg/agent/testdata/AKSUbuntu1804+Containerd+Kubenet/line200.sh [hunk 3]
```diff
@@ -1,9 +0,0 @@
-[Unit]
```
Comment (cpuguy83) [inline pkg/agent/testdata/AKSUbuntu1804+Containerd+Kubenet/line200.sh:1]: What are all these `line200.sh` changes?

**Review by cpuguy83 — COMMENTED** [review]
How come the diff is so big? I'd expect to see `s/io.containerd.runtime.v1.linux/io.containerd.runc.v2/`

File: pkg/agent/testdata/AKSUbuntu1804+Containerd+Kubenet/line200.sh (hunk 3 above)
Comment (eriksywu) [reply pkg/agent/testdata/AKSUbuntu1804+Containerd+Kubenet/line200.sh:1]: We have a bunch of pre-defined node provisioning scenarios (i.e containerd + kubenet cni) and our make generate target runs templating engine to generate the cloudinit files - this is mostly just a way to sanity check that the outputs are correct
not 100% sure the output files are named like that, tbh

Comment (eriksywu) [reply pkg/agent/testdata/AKSUbuntu1804+Containerd+Kubenet/line200.sh:1]: for the most part it is `s/io.containerd.runtime.v1.linux/io.containerd.runc.v2/` apart from the generated test data like this file?
Not sure if I need to specify default_runtime but it seems like it's better to be safe?

the nvidia-gpu-runtime configs need a bit more lines of config to specify the binary location  so I've completely separate the templating to make it more readable

File: parts/linux/cloud-init/nodecustomdata.yml (hunk 2 above)
Comment (eriksywu) [reply parts/linux/cloud-init/nodecustomdata.yml:288]: Has runtime_engine been deprecated?
I don't see it in the readme: https://github.com/containerd/containerd/blob/master/docs/cri/config.md

Comment (cpuguy83) [reply parts/linux/cloud-init/nodecustomdata.yml:288]: I messed with this today, it only applies to `io.containerd.runtime.v1.linux`... this config is fun.

File: parts/linux/cloud-init/nodecustomdata.yml [hunk 4]
```diff
@@ -315,49 +315,53 @@ write_files:
...
-        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.untrusted_workload_runtime.options]
+        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.untrusted.options]
           BinaryName = "/usr/bin/nvidia-container-runtime"
-        {{ else}}
+        {{- else}}
         default_runtime_name = "runc"
         [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc]
           runtime_type = "io.containerd.runc.v2"
+        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc.options]
+          BinaryName = "/usr/bin/runc"
```
Comment (cpuguy83) [inline parts/linux/cloud-init/nodecustomdata.yml:343]: `BinaryName` is not necessary in this case, unless you want to be explicit.

**Review by cpuguy83 — COMMENTED** [review]
Hard to review the template but it looks pretty good.
Minor comment on binary names for runc, otherwise LGTM.

**Pull request merged**